src/ucp/wireup/select.c

```diff
@@ -1176,14 +1200,46 @@ ucp_wireup_add_rma_bw_lanes(const ucp_wireup_select_params_t *select_params,
     bw_info.usage             = UCP_WIREUP_LANE_USAGE_RMA_BW;
 
     UCS_STATIC_ASSERT(UCS_MEMORY_TYPE_HOST == 0);
-    for (mem_type = UCS_MEMORY_TYPE_HOST; mem_type < UCS_MEMORY_TYPE_LAST; mem_type++) {
-        if (!context->mem_type_access_tls[mem_type]) {
-            continue;
+    /* First checked RNDV mode has to be a mode specified in config */
+    ucs_assert(rndv_modes[0] == context->config.ext.rndv_mode);
+
+    /* RNDV protocol can't mix different schemes, i.e. wireup has to
+     * select lanes with the same iface flags depends on a requested
+     * RNDV scheme.
+     * First of all, try to select lanes with RNDV scheme requested
+     * by user. If no lanes were selected and RNDV scheme in the
+     * configuration is AUTO, try other schemes. */
+    for (i = 0; i < ucs_array_size(rndv_modes); i++) {
+        /* Remove the previous iface RMA flags */
+        bw_info.criteria.remote_iface_flags &= ~iface_rma_flags;
+        bw_info.criteria.local_iface_flags  &= ~iface_rma_flags;
+
+        iface_rma_flags = ucp_wireup_get_rma_bw_iface_flags(rndv_modes[i]);
+
+        /* Set the new iface RMA flags */
+        bw_info.criteria.remote_iface_flags |= iface_rma_flags;
+        bw_info.criteria.local_iface_flags  |= iface_rma_flags;
+
+        added_lanes = 0;
+
+        for (mem_type = UCS_MEMORY_TYPE_HOST;
```

Comment:
would be reasonable to have `UCS_STATIC_ASSERT(UCS_MEMORY_TYPE_HOST == 0);` above this

Reply:
fixed